File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/cot/Java/C++/s196249385_comments_removed.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

long highestOneBit(long x) {
    x |= (x >> 1);
    x |= (x >> 2);
    x |= (x >> 4);
    x |= (x >> 8);
    x |= (x >> 16);
    return x - (x >> 1);
}

void solve() {
    int n, k;
    cin >> n >> k;

    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }
    
    vector<long> cumsum(n + 1, 0);
    for (int i = 0; i < n; i++) {
        cumsum[i + 1] = cumsum[i] + a[i];
    }

    vector<long> b;
    b.reserve(n * (n + 1) / 2);
    long max = 0;
    for (int i = 0; i < n; i++) {
        for (int j = i; j < n; j++) {
            b.push_back(cumsum[j + 1] - cumsum[i]);
            max = std::max(max, cumsum[j + 1] - cumsum[i]);
        }
    }

    long mask = highestOneBit(max);
    while (mask > 0) {
        vector<long> tmp;
        for (long e : b) {
            if ((e & mask) != 0) {
                tmp.push_back(e);
            }
        }
        
        if (tmp.size() >= k) {
            b = tmp;
        }
        
        mask >>= 1;
    }
    
    if (b.size() < k) {
        cout << 0 << endl;
    } else {
        long ans = -1L;
        for (long e : b) {
            ans &= e;
        }
        
        cout << ans << endl;
    }
}

int main() {
    solve();
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/Python/s196249385_comments_removed.py
from typing import List

class Main:
    sc = None
    pr = None

    @staticmethod
    def solve():
        n = Main.sc.nextInt()
        k = Main.sc.nextInt()

        a = [0] * n
        for i in range(n):
            a[i] = Main.sc.nextInt()

        cumsum = [0] * (n + 1)
        for i in range(n):
            cumsum[i + 1] = cumsum[i] + a[i]

        b = []
        max_val = 0
        for i in range(n):
            for j in range(i, n):
                b.append(cumsum[j + 1] - cumsum[i])
                max_val = max(max_val, cumsum[j + 1] - cumsum[i])

        mask = 1 << (max_val.bit_length() - 1)
        while mask > 0:
            tmp = [e for e in b if (e & mask) != 0]
            if len(tmp) >= k:
                b = tmp
            mask >>= 1

        if len(b) < k:
            Main.pr.println(0)
        else:
            ans = -1
            for e in b:
                ans &= e
            Main.pr.println(ans)

    @staticmethod
    def main():
        Main.sc = Scanner()
        Main.pr = Printer()
        Main.solve()
        Main.pr.close()
        Main.sc.close()

class Scanner:
    def __init__(self):
        self.input = input

    def nextInt(self):
        return int(self.input())

class Printer:
    def __init__(self):
        self.output = print

    def println(self, value):
        self.output(value)

    def close(self):
        pass

Main.main()
# End of Code
